gulpfile.js

```diff
@@ -3,94 +3,110 @@ var gulp = require('gulp')
+function browserSync(done) {
+    browsersync.init({
+        server: {
+            baseDir: "./dist/"
+        },
+        port: 3000
+    });
+    done();
+}
+
+// BrowserSync Reload
+function browserSyncReload(done) {
+    browsersync.reload();
+    done();
+}
+
+// WATCH TASKS END
+
+// BUILD TASKS START
+
+function cleanDist() {
+    return del(['dist/*']);
+}
+
+function copyfiles() {
+    return gulp.src('src/client/images/**')
+    .pipe(gulp.dest('dist/client/images'));
+}
+
+function css() {
+    return gulp
+        .src('src/client/css/*.css')
+        .pipe(postcss([cssnano()]))
+        .pipe(rename({suffix: '.min.css'}))
+        .pipe(gulp.dest('dist/client/css'));
+}
+
+function scriptsLint() {
+    return gulp
+        .src(['src/client/*.es6.js','!node_modules/**'])
+        .pipe(eslint())
+        .pipe(eslint.format())
+        .pipe(eslint.failAfterError())
+}
+
+function babeltask() {
     return gulp.src('src/client/js/app.es6.js')
-        .pipe(babel({presets: ['es2015']}))
+        .pipe(babel({presets: ["@babel/preset-env"]}))
         .pipe(rename('app.js'))
         .pipe(gulp.dest('src/client/js'))
-})
+}
 
-gulp.task('usemin', function() {
+function scripts() {
+    return (
+        gulp
+            .src('src/client/js/app.js')
+            .pipe(sourcemaps.init())
+            .pipe(concat('app.min.js'))
+            .pipe(uglify())
+            .pipe(sourcemaps.write('./'))
+            .pipe(gulp.dest('dist/client/js'))
+    );
+}
+
+function useminfiles() {
     return gulp.src('src/index.html')
         .pipe(usemin({
-            //assetsDir: 'images',
-            css: [cssmin(), 'concat'],
+            css: [(postcss([cssnano()])), 'concat'],
             js: [uglify(), 'concat']
         }))
         .pipe(gulp.dest('dist'));
-});
+}
 
-// copy images to dist/
-gulp.task('copyfiles', function() {
-    gulp.src('src/client/images/**')
-    .pipe(gulp.dest('dist/client/images'));
-});
+// BUILD TASKS END
 
-gulp.task('lint', function () {
-    return gulp.src(['src/client/*.es6.js','!node_modules/**'])
-        .pipe(eslint())
-        .pipe(eslint.format())
-        .pipe(eslint.failAfterError())
-})
-
-// Concat & Minify JS
-gulp.task('uglify', function(){
-  return gulp.src('src/client/js/app.js')
-        .pipe(sourcemaps.init())
-        .pipe(concat('app.min.js'))
-        //.pipe(rename('app.min.js'))
-        .pipe(uglify())
-        //.pipe(uglify( {compress: {drop_debugger:false} }))
-        .pipe(sourcemaps.write('./'))
-        .pipe(gulp.dest('dist/client/js'))
-})
-
-
-gulp.task('cssmin', function () {
-    return gulp.src('src/client/css/*.css')
-        .pipe(cssmin())
-        .pipe(rename({suffix: '.min.css'}))
-        .pipe(gulp.dest('dist/client/css'));
-});
-
-
-// remove anything in dist
-gulp.task('clean', function() {
-    return gulp.src(['dist/*'], {read: false})
-        .pipe(clean());
-});
-
-gulp.task('clean-es5', function () {
+function cleanAppJs() {
```

Comment:
The trick with implementing this is that when we delete the old .js file, the `watch` task detects the change and triggers a build... which creates a new `app.js` that is subsequently deleted & detected by the `watch` task, etc etc. I can tell the previous dev ran into this issue because they commented out a line in their original code which would've triggered this same loop.

I believe the solution to implement is to send the app.js into a build folder, which is not watched, and is then deleted. `useMin` will need to receive a relative path option to check for `app.js` in the build folder rather than the relative path.